### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by daun**
**Description**

- Create unit tests for our custom queue implementation
- Create functional tests for the plugin and all options
- Add workflows to run both in CI

**Checks**

- [x] The PR is submitted to the `master` branch
- [x] The code was linted before pushing (`npm run lint`)
- [x] All tests are passing (`npm run test`)
- [x] New or updated tests are included
- [ ] ~~The documentation was updated as required~~

**Required changes**

- Update to and require swup 4.5 to get `touchstart` to work properly 🤔
- Use new argument order for hook calls
- Wrap preload function in `page:preload` hook

### CODE REVIEW

**Comment by hirasso** [comment]
Besutiful! Is the changed argument order a breaking change?

**Comment by daun** [comment]
@hirasso No, swup currently accepts both old and new hook argument order. This change just updates to the current canonical order.

**Comment by daun** [comment]
@hirasso Ah, I misread your question. The updated arguments are for the call to `swup.hooks.call()`. The arguments of the `page:preload` hook itself haven't changed.

**Comment by hirasso** [comment]
Ah ok 👌

File: tests/config/serve.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "public": "tests/fixtures",
+  "cleanUrls": false,
+  "redirects": [
```
Comment (hirasso) [inline tests/config/serve.json:4]: do we need the redirects directive here?

File: tests/config/vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+	test: {
+		environment: 'jsdom',
+		include: [
+			'tests/unit/**/*.test.ts'
+		],
+		setupFiles: [
+			'tests/config/vitest.setup.ts'
```
Comment (hirasso) [inline tests/config/vitest.config.ts:15]: Leaving this here so we can quickly use it should we need it sometime later?

File: tests/config/vitest.setup.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { vi } from 'vitest';
```
Comment (hirasso) [inline tests/config/vitest.setup.ts:1]: Unused right now – leaving it here for future usage?

**Review by hirasso — APPROVED** [review]

**Comment by hirasso** [comment]
I guess all my comments about unused stuff also apply to fragment plugin :) ...no big deal, we can totally leave it in.

File: tests/config/serve.json (hunk 1 above)
Comment (daun) [reply tests/config/serve.json:4]: We currently don't test the expected behavior for redirected requests, but I think we should. For example, 302 redirects probably shouldn't preload, but we don't know the exact status code, so we should discard all redirected responses anyway.

File: tests/config/vitest.config.ts (hunk 2 above)
Comment (daun) [reply tests/config/vitest.config.ts:15]: I think so — let's merge this for now, and then do a more thorough review when comparing all plugins with e2e tests.

**Pull request merged**